### PR TITLE
CVD for steel grades

### PR DESCRIPTION
**Pull request by Drew4484**
#### Summary
Content "Allows various steel grades to receive CVD coating"

#### Purpose of change
As more weapons are made with grades of steel, players may want to apply CVD coating to them.

#### Describe the solution
Expands the list of materials allowed for CVD.

#### Describe alternatives you've considered
None.

#### Testing
WIP

#### Additional context
None.

### CODE REVIEW

File: src/iexamine.cpp [hunk 1]
```diff
@@ -288,13 +293,25 @@ bool iexamine::harvestable_now( const tripoint &examp )
 void iexamine::cvdmachine( Character &you, const tripoint & )
 {
     // Select an item to which it is possible to apply a diamond coating
-    item_location loc = g->inv_map_splice( []( const item & e ) {
-        return ( e.is_melee( damage_type::CUT ) || e.is_melee( damage_type::STAB ) ) &&
-               e.made_of( material_steel ) &&
-               !e.has_flag( flag_DIAMOND ) && !e.has_flag( flag_NO_CVD );
+    std::array<std::reference_wrapper<const material_id>, 6> steels = {
+        material_steel,
```
Comment (AtomicFox556) [inline src/iexamine.cpp:297]: While you're at it, I think iron could work here as well.

Comment (Drew4484) [reply src/iexamine.cpp:297]: I'm working towards removing any crafted iron weapons, by the time a player can work iron it's hardly any more difficult to make steel and the result is more worth the CVD process.

File: src/iexamine.cpp [hunk 2]
```diff
@@ -288,12 +288,20 @@ bool iexamine::harvestable_now( const tripoint &examp )
...
     // Select an item to which it is possible to apply a diamond coating
+    std::array<std::reference_wrapper<const material_id>, 6> steels = {
+        material_steel,
+        material_lc_steel,
+        material_mc_steel,
+        material_hc_steel,
+        material_ch_steel,
+        material_qt_steel,
+    };
     item_location loc = g->inv_map_splice( []( const item & e ) {
```
Comment (BrettDong) [inline src/iexamine.cpp:299]: Need to capture `steels` in the lambda expression
```cpp
    item_location loc = g->inv_map_splice( [&steels]( const item & e ) {
```

Comment (Drew4484) [reply src/iexamine.cpp]: TY!

Comment (BrettDong) [reply src/iexamine.cpp]: It needs to be with an ampersand so it is captured by reference:

```cpp
[&steels]( const item & e ) { ... }
```

The meaning is different without the ampersand.
```cpp
[steels]( const item & e ) { ... }
```
It copies `steels` every time.

**Comment by Drew4484** [comment]
Resolved by #60321

**Pull request closed without merging**